src/Components/ControlBar/ControlBar.js

```diff
@@ -0,0 +1,245 @@
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+import Checkbox from "@material-ui/core/Checkbox";
+import CircularProgress from "@material-ui/core/CircularProgress";
+import Divider from "@material-ui/core/Divider";
+import Typography from "@material-ui/core/Typography";
+
+const useStyles = makeStyles(theme => ({
+    root: {
+        display: "flex",
+        flexFlow: "row noWrap"
+    },
+    content: {
+        padding: theme.spacing(1),
+        flex: "1 1 100%",
+        height: "100%"
+    },
+    column: {
+        flex: "1 1 auto",
+        display: "flex",
+        flexFlow: "column noWrap"
+    },
+    row: {
+        width: "100%",
+        display: "flex"
+    },
+    pad: {
+        padding: theme.spacing(1)
+    },
+    open: {
+        maxWidth: 500,
+        height: "100%",
+        backgroundColor: theme.palette.primary.main
+    }
+}));
+
+function ControlBar({
+    settings,
+    instructions,
+    updateLanguage,
+    updateIsAdaptive
+}) {
+    const classes = useStyles(Theme);
+
+    const [loading, setLoading] = useState(false);
+    const [settingsOpen, setSettingsOpen] = useState(false);
+    const [instructionsOpen, setInstructionsOpen] = useState(false);
+
+    const [languages, setLanguages] = useState([]);
+    const [language, setLanguage] = useState("");
+
+    const [isAdaptive, setIsAdaptive] = useState(false);
+
+    function handleSettingsClick() {
+        setInstructionsOpen(false);
+        setSettingsOpen(prevState => !prevState);
+    }
+    function handleInstructionsClick() {
+        setSettingsOpen(false);
+        setInstructionsOpen(prevState => !prevState);
+    }
+
+    function handleAdaptiveChange() {
+        setIsAdaptive(prevState => !prevState);
+    }
+    async function getLanguages() {
+        setLoading(true);
+        return Promise.resolve(ListLanguages())
+            .then(ls => setLanguages(ls))
+            .finally(() => setLoading(false));
+    }
+
+    useEffect(() => {
+        getLanguages().finally();
+    }, []);
+
+    useEffect(() => {
+        if (language && updateLanguage) {
+            updateLanguage(language);
+        }
+    }, [language, updateLanguage]);
+    useEffect(() => {
+        if (updateIsAdaptive) {
+            updateIsAdaptive(isAdaptive);
+        }
+    }, [isAdaptive, updateIsAdaptive]);
+    return (
+        <div className={clsx(classes.root)}>
+            <div className={clsx(classes.column)} style={{ maxWidth: 225 }}>
+                {updateLanguage && (
+                    <div className={clsx(classes.row)}>
+                        <div
+                            className={clsx(classes.pad)}
+                            style={{
+                                marginTop: Theme.spacing(1),
+                                padding: Theme.spacing(2),
+                                backgroundColor: Theme.palette.primary.dark,
+                                borderTopRightRadius: "100%",
```

Comment:
By making percentages instead of pixel values, we get elongated, elliptical corners for divs that aren't square. (percentage of width and height aren't the same on the longer divs)
Just making each 100 [px] instead gives even rounded corners,